sdk/appconfiguration/app-configuration/src/secretReference.ts

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { errorMessageForUnexpectedSetting } from "./internal/helpers";
+import { JsonSecretReferenceValue } from "./internal/jsonModels";
+import { ConfigurationSetting, ConfigurationSettingParam } from "./models";
+
+/**
+ * content-type for the secret reference.
+ */
+export const secretReferenceContentType =
+  "application/vnd.microsoft.appconfig.keyvaultref+json;charset=utf-8";
+
+/**
+ * Necessary fields for updating or creating a new secret reference.
+ */
+export interface SecretReferenceValue {
+  /**
+   * Id for the secret reference.
+   */
+  secretId: string;
+}
+
+/**
+ * @internal
+ */
+export const SecretReferenceHelper = {
+  /**
+   * Takes the ConfigurationSetting and returns the SecretReference.
+   */
+  fromConfigurationSetting: (
+    setting: ConfigurationSetting
+  ): ConfigurationSetting<SecretReferenceValue> => {
+    if (!isSecretReference(setting)) {
+      throw TypeError(errorMessageForUnexpectedSetting(setting.key, "SecretReference"));
+    }
+
+    const jsonSecretReferenceValue = JSON.parse(setting.value) as JsonSecretReferenceValue;
+
+    const featureflag: ConfigurationSetting<SecretReferenceValue> = {
+      ...setting,
+      value: { secretId: jsonSecretReferenceValue.uri }
+    };
+    return featureflag;
+  },
+  /**
+   * Takes the SecretReference (JSON) and returns a ConfigurationSetting (with the props encodeed in the value).
+   */
+  toConfigurationSettingParam: (
+    secretReference: ConfigurationSettingParam<SecretReferenceValue>
+  ): ConfigurationSettingParam => {
+    if (!secretReference.value) {
+      throw new TypeError(`SecretReference has an unexpected value - ${secretReference.value}`);
+    }
+
+    const jsonSecretReferenceValue: JsonSecretReferenceValue = {
+      uri: secretReference.value.secretId
+    };
+
+    const configSetting = {
+      ...secretReference,
+      value: JSON.stringify(jsonSecretReferenceValue)
+    };
+    return configSetting;
+  }
+};
+
+/**
+ * Takes the ConfigurationSetting as input and returns the ConfigurationSetting<SecretReferenceValue> by parsing the value string.
+ */
+export const parseSecretReference = SecretReferenceHelper.fromConfigurationSetting;
```

Comment:
why are we also exporting this method like this in addition the object form above?

Reply:
That object is not exported to the users, only this method is being exported.